rust/lance/src/index.rs

```diff
@@ -108,7 +108,13 @@ pub(crate) async fn remap_index(
 
     match generic.index_type() {
         it if it.is_scalar() => {
-            let new_store = LanceIndexStore::from_dataset(dataset, &new_id.to_string());
+            let new_store = match it {
```

Comment:
Repetitive, but would also help to have this comment here.
```suggestion
            // The BTree index implementation leverages the legacy format's batch offset,
            // which has been removed from new format, so keep using the legacy format for now.
            let new_store = match it {
```

Reply:
Yes I copy the comments on purpose, let's keep it for now, can remove them after btree in v2 format ready